documentation/dsi-integration.md

```diff
@@ -0,0 +1,171 @@
+
+```
+DFE_SIGN_IN_IDENTIFIER=
+DFE_SIGN_IN_PASSWORD=
+DFE_SIGN_IN_SECRET=
+DFE_SIGN_IN_SERVICE_ACCESS_ROLE_ID=
+DFE_SIGN_IN_SERVICE_ID=
+```
+
+## DSI Contingency Fallback Playbook
+
+### Background
+
+We rely on DfE Sign In (DSI) to provide our authentication and authorisation.
+
+This is a contingency plan for when DSI has an outage. The fallback authentication method relies on the data we have on users from our nightly job UpdateDsiUsersInDbJob: specifically, the email addresses and the organisation URNs.
+
+It replaces the DSI sign in method with one whereby the user is prompted to enter their email address, and clicks a unique login link. This login link works only once and expires after a configurable time.
+
+### How to use
+
+Here are the steps to follow to use our contingency fallback sign-in method.
+
+First, decide whether to switch on the fallback authentication. This call should be made by the Product Owner/Manager if they are available.
+   - An alternative to this fallback sign in method could be replacing DSI with a notice to users telling them that they can't access the service. To do this, we would reinstate the environment variable and code which was deleted in commit [`🔥 Remove FEATURE_SIGN_IN_ALERT flag`](https://github.com/DFE-Digital/teaching-vacancies/commit/bc12fb9808c955f86cd87e62648a76786516e2c3).
+
+### Enabling Authentication Fallback
+
+Switch on the fallback authentication by setting the environment variable `AUTHENTICATION_FALLBACK` to `true`.
+
+The following method is the fastest, but requires `SpaceDeveloper` permission on `production`, and only persists until the next [Automated Deployment to staging and production](/deployments.md#build-and-deploy-to-staging-and-production---github-actions)
+
+- Block all deployments to `production` by requesting in the Slack channel `#tv_engineering`
+- Log in to GOV.UK PaaS (with `cf login --sso`). You will need a [Passcode](https://login.london.cloud.service.gov.uk/passcode)
+- Select the `teaching-vacancies-production` space from the menu, or Switch to the production space:
+```bash
+cf target -s teaching-vacancies-production
+```
+
+- Get the current setting with:
+```bash
+cf env teaching-vacancies-production | grep "AUTHENTICATION_FALLBACK:"
+```
+
+- Update the environment variable with the [set-env](http://cli.cloudfoundry.org/en-US/v7/set-env.html) command:
+```bash
+cf set-env teaching-vacancies-production AUTHENTICATION_FALLBACK true
+```
+
+- [Restage](http://cli.cloudfoundry.org/en-US/v7/restage.html) the app to pick up the updated configuration:
+```bash
+cf restage teaching-vacancies-production --strategy rolling
+```
+
+Changing the environment variable within the `<env>_app_env.yml` is a more persistent change, and allows deployments to production to continue:
```

Comment:
I think we should emphasize this as the main solution. The pipeline is quick enough to enable this.
The manual option should only be used in panic-mode.

Reply:
Agreed - let's detail this method before the other one.

Reply:
Have made it more explicit that the manual workaround is only there for completeness